### PR TITLE
RUBY-2997 Allow loading schema map from file

### PR DESCRIPTION
**Pull request by comandeo**

### CODE REVIEW

**Review by Neilshweky — APPROVED** [review]
Awesome!

File: docs/reference/client-side-encryption.txt [hunk 1]
```diff
@@ -919,6 +919,22 @@ When you intend to use your schema map, convert it to a Ruby ``Hash`` using the
   :drivers:`Specify Encrypted Fields Using JSON Schema</security/client-side-field-level-encryption-guide/#c-specify-encrypted-fields-using-json-schema>`,
   :manual:`Automatic Encryption Rules</reference/security-client-side-automatic-json-schema/>`
 
+``:schema_map_file``
```
Comment (p-mongo) [inline docs/reference/client-side-encryption.txt:922]:
```suggestion
``:schema_map_path``
```

File: docs/reference/client-side-encryption.txt [hunk 2]
```diff
@@ -919,6 +919,22 @@ When you intend to use your schema map, convert it to a Ruby ``Hash`` using the
...
+~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to load schema map from a file. Prepare the schema as
```
Comment (p-mongo) [inline docs/reference/client-side-encryption.txt:925]:
```suggestion
It is also possible to load schema map from a file. Prepare the schema map as
```

File: docs/reference/client-side-encryption.txt [hunk 3]
```diff
@@ -919,6 +919,22 @@ When you intend to use your schema map, convert it to a Ruby ``Hash`` using the
...
+described above, save it to file, and then pass path to the file using
+``:schema_map_file`` option.
```
Comment (p-mongo) [inline docs/reference/client-side-encryption.txt:927]:
```suggestion
``:schema_map_path`` option:
```

File: docs/reference/client-side-encryption.txt [hunk 4]
```diff
@@ -919,6 +919,22 @@ When you intend to use your schema map, convert it to a Ruby ``Hash`` using the
...
+
+.. code-block:: ruby
+
+  Mongo::Client.new(['localhost:27017],
+    auto_encryption_options: {
+      schema_map_file: '/path/to/your/file.json',
```
Comment (p-mongo) [inline docs/reference/client-side-encryption.txt:933]:
```suggestion
      schema_map_path: '/path/to/your/file.json',
```

**Review by p-mongo — APPROVED** [review]
I prefer to use "path" to refer to paths, and "file" to refer to file objects, which in Ruby would be the IO ones.

**Comment by comandeo** [comment]
> I prefer to use "path" to refer to paths, and "file" to refer to file objects, which in Ruby would be the IO ones.

Changed accordingly.

**Pull request merged**